src/App.tsx

```diff
@@ -750,6 +742,13 @@ export default function App() {
           </ErrorBoundary>
         </div>
       </div>
+    </>
+  );
+
+  return (
+    <StyledEngineProvider injectFirst>
+      <CssBaseline />
+      <UserProvider data={{ user, reloadUser }}>{getContent()}</UserProvider>
```

Comment:
ideally we fetch the user within the provider layer, for now i'm keeping the existing fetching sequence just not to break anything. will revisit this later.